### PR TITLE
FINAL CHANGES

### PR DESCRIPTION
**Pull request by AndrewNQN**
Screens: Refactored and Stylized

Use Cases: All plugged into their respective screens (not Move Card, sorry, Derrick)

Fail Views: All fail view presents an error message with corresponding information. The error messages have been changed for several use cases to be consistent with each other. Updated test cases to align with these changes

Documentation: Made sure all documentation was up-to-standard.

Java Warnings: Got rid of most Java warnings (unused import statements, redundant stuff..)

Flashcard Creation: Flashcard Factory no longer uses MAGIC NUMBERS. New Enum State was added. Changes to CreateCard and Import Use cases were made to reflect this change.

Overall, functionality is all there. All that is left is a few more style changes to the running quiz process. Otherwise, everything is complete and looks good to hand in. Documentation is missing in a few areas, but that will be assigned after today's meeting. Use Case testing is not fully there yet, so everyone may need to spend a bit more time creating a few more test cases.

### CODE REVIEW

File: src/main/java/createQuiz/CreateQuizInteractor.java [hunk 1]
```diff
@@ -44,11 +44,11 @@ public void createQuiz(CreateQuizInputData inputData) {
         // Fail views
         if (Quiz.getTracker().containsKey(quizName)) {
             CreateQuizOutputData outputData = new CreateQuizOutputData(null,
-                    "This quiz name exists already. Failed to create new quiz.");
+                    "Quiz name already exists. Please choose a different name.");
```
Comment (aldob02) [inline src/main/java/createQuiz/CreateQuizInteractor.java:47]: better error message

File: src/main/java/data/DeckPathCleaner.java [hunk 2]
```diff
@@ -1,5 +1,13 @@
+/*
```
Comment (aldob02) [inline src/main/java/data/DeckPathCleaner.java:1]: thanks for putting this in for me

File: src/main/java/editCard/EditCardController.java [hunk 3]
```diff
@@ -1,22 +1,20 @@
-package editCard;
 /*
  * Author: Albert Yan
  * Date: October 31/22
  * The edit card controller for edit card systems, receives the arguments required to perform an edit card action
```
Comment (aldob02) [inline src/main/java/editCard/EditCardController.java:4]: think this needs to be a javadoc right above the class name (to be consistent with other files)

File: src/main/java/editCard/EditCardController.java [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
  */
 
-import entities.Flashcard;
-
-import java.util.List;
-
-/**
+/*
  * This file contains an implementation of the EditCardController class
```
Comment (aldob02) [inline src/main/java/editCard/EditCardController.java:8]: remove this comment

File: src/main/java/editCard/EditCardInputBoundary.java [hunk 5]
```diff
@@ -1,14 +1,15 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * This interface allows cards to be edited by higher level classes
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardInputBoundary.java:6]: see comment in the EditCardController class

File: src/main/java/editCard/EditCardInputData.java [hunk 6]
```diff
@@ -1,19 +1,20 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * This class is a container for the input data related to edit card systems,
  * it is created by the controller and sent to the interactor
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardInputData.java:7]: see comment in the EditCardController class

File: src/main/java/editCard/EditCardInteractor.java [hunk 7]
```diff
@@ -1,19 +1,20 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * The use case interactor for editing cards given an input boundary
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardInteractor.java:6]: see comment in the EditCardController class

File: src/main/java/editCard/EditCardOutputBoundary.java [hunk 8]
```diff
@@ -1,15 +1,16 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * The output boundary for edit card systems
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardOutputBoundary.java:6]: see comment in the EditCardController class

File: src/main/java/editCard/EditCardOutputData.java [hunk 9]
```diff
@@ -1,27 +1,28 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * The output data class for edit card systems, packages data related to output together to be sent to viewmodel
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardOutputData.java:6]: see comment in the EditCardController class

File: src/main/java/editCard/EditCardPresenter.java [hunk 10]
```diff
@@ -1,18 +1,20 @@
-package editCard;/*
+/*
  * Author: Albert Yan
  * Date: October 31/22
  * The presenter class for edit card systems
  */
```
Comment (aldob02) [inline src/main/java/editCard/EditCardPresenter.java:6]: see comment in the EditCardController class

File: src/main/java/editQuiz/EditQuizOutputData.java [hunk 11]
```diff
@@ -9,16 +9,27 @@
```
Comment (aldob02) [inline src/main/java/editQuiz/EditQuizOutputData.java:9]: for consistency put the description above the class name as a javadoc

File: src/main/java/createQuiz/CreateQuizInteractor.java (hunk 1 above)
Comment (AndrewNQN) [reply src/main/java/createQuiz/CreateQuizInteractor.java:47]: Better as in a good job or I could do better?

Comment (aldob02) [reply src/main/java/createQuiz/CreateQuizInteractor.java:47]: as in good job

File: src/main/java/data/DeckPathCleaner.java (hunk 2 above)
Comment (AndrewNQN) [reply src/main/java/data/DeckPathCleaner.java:1]: No Problem

File: src/main/java/moveCard/MoveCardController.java [hunk 12]
```diff
@@ -1,13 +1,10 @@
-package moveCard;
 /*
  * Author: Derrick Cho
  * Date: November 13/22
  * Description:
```
Comment (aldob02) [inline src/main/java/moveCard/MoveCardController.java:4]: remove blank description (already included as javadoc)

File: src/main/java/editCard/EditCardController.java (hunk 3 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardController.java:4]: Agreed. That can be changed after presentations tomorrow

File: src/main/java/moveCard/MoveCardInputBoundary.java [hunk 13]
```diff
@@ -1,9 +1,15 @@
+/*
+ * Author: Derrick Cho
+ * Date: November 13/22
+ * Description:
```
Comment (aldob02) [inline src/main/java/moveCard/MoveCardInputBoundary.java:4]: remove blank description (already included as javadoc above interface name)

File: src/main/java/moveCard/MoveCardInputData.java [hunk 14]
```diff
@@ -1,7 +1,10 @@
-package moveCard;
+/*
+ * Author: Derrick Cho
+ * Date: November 13/22
+ * Description:
```
Comment (aldob02) [inline src/main/java/moveCard/MoveCardInputData.java:4]: remove blank description

File: src/main/java/editCard/EditCardController.java (hunk 3 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardController.java:4]: Done.

File: src/main/java/editCard/EditCardController.java (hunk 4 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardController.java]: Done.

File: src/main/java/editCard/EditCardInputBoundary.java (hunk 5 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardInputBoundary.java:6]: Done.

File: src/main/java/editCard/EditCardInteractor.java (hunk 7 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardInteractor.java]: Done.

File: src/main/java/editCard/EditCardOutputData.java (hunk 9 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardOutputData.java:6]: No way youre calling out every comment from the edit card use case, but Done.

File: src/main/java/screens/CardScreen.java [hunk 15]
```diff
@@ -6,15 +6,11 @@
...
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
-import static javax.swing.JOptionPane.showMessageDialog;
-
 public class CardScreen extends JPanel {
```
Comment (aldob02) [inline src/main/java/screens/CardScreen.java:14]: if you've got time maybe comment on what's going on?

File: src/main/java/editCard/EditCardPresenter.java (hunk 10 above)
Comment (AndrewNQN) [reply src/main/java/editCard/EditCardPresenter.java]: Done.

File: src/main/java/moveCard/MoveCardController.java (hunk 12 above)
Comment (AndrewNQN) [reply src/main/java/moveCard/MoveCardController.java:4]: Done.

File: src/main/java/screens/CardScreen.java (hunk 15 above)
Comment (aldob02) [reply src/main/java/screens/CardScreen.java:14]: also javadoc

File: src/main/java/editQuiz/EditQuizOutputData.java (hunk 11 above)
Comment (AndrewNQN) [reply src/main/java/editQuiz/EditQuizOutputData.java:9]: Another day.

File: src/main/java/moveCard/MoveCardInputBoundary.java (hunk 13 above)
Comment (AndrewNQN) [reply src/main/java/moveCard/MoveCardInputBoundary.java:4]: Done.

File: src/main/java/moveCard/MoveCardInputData.java (hunk 14 above)
Comment (AndrewNQN) [reply src/main/java/moveCard/MoveCardInputData.java:4]: Done.

File: src/main/java/screens/CardScreen.java (hunk 15 above)
Comment (AndrewNQN) [reply src/main/java/screens/CardScreen.java:14]: Removed unnecessary import statements. We did not end up using that

File: src/main/java/screens/DeckScreen.java [hunk 16]
```diff
@@ -9,12 +9,8 @@
 
 import javax.swing.*;
 import java.awt.*;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
 import java.util.ArrayList;
 
-import static javax.swing.JOptionPane.showMessageDialog;
-
 public class DeckScreen extends JPanel {
```
Comment (aldob02) [inline src/main/java/screens/DeckScreen.java:14]: see comment in CardScreen

File: src/main/java/screens/View.java [hunk 17]
```diff
@@ -14,12 +14,9 @@
...
 import runQuiz.RunQuizController;
-import runQuiz.RunQuizInteractor;
 
 import javax.swing.*;
 import java.awt.*;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
 import java.util.List;
 
 public class View implements ViewBoundary {
```
Comment (aldob02) [inline src/main/java/screens/View.java:22]: see comment in CardScreen

File: src/main/java/screens/ViewModel.java [hunk 18]
```diff
@@ -9,7 +9,6 @@ public class ViewModel {
     //Required Parameter(s):
```
Comment (aldob02) [inline src/main/java/screens/ViewModel.java:9]: see comment in CardScreen

File: src/test/java/editQuiz/EditQuizInteractorTest.java [hunk 19]
```diff
@@ -53,7 +53,7 @@ void deckSetup(){
 
     @BeforeEach
     void quizSetup(){
-        List<Deck> quizDecks = new ArrayList<Deck>();
+        List<Deck> quizDecks = new ArrayList<>();
```
Comment (aldob02) [inline src/test/java/editQuiz/EditQuizInteractorTest.java:56]: nice catch

File: src/main/java/screens/DeckScreen.java (hunk 16 above)
Comment (AndrewNQN) [reply src/main/java/screens/DeckScreen.java:14]: see comment in CardScreen

**Review by aldob02 — CHANGES_REQUESTED** [review]
Overall good changes, error messages are more descriptive, and there's more consistency between everything.  I've nitpicked on every lack of consistency I noticed.

File: src/main/java/screens/View.java (hunk 17 above)
Comment (AndrewNQN) [reply src/main/java/screens/View.java:22]: see comment in CardScreen

**Review by aldob02 — APPROVED** [review]

**Review by JamieJChew — APPROVED** [review]
Wow, lot's of changes! I like that you changed all of the card types from being just integers (1, 2, 3) to using an Enum, it's a lot more readable this way. I also like how a lot of the code is more compact now and less clunky, especially in CardScreen.java and DeckScreen.java. This is looking pretty good. 👍

**Review by derrickcho3715 — APPROVED** [review]
All good changes. I think the removal of unused import statements and more uniform styling and documentation result in a much more compact and readable code. The updated error messages are also much better. Good work.

**Review by tsoisze2 — APPROVED** [review]
Thanks for fixing the blank lines warnings for these classes. I think the change of getting card type  from taking in an integer in the create card use case is better than before. Also the documentations are consistent across use cases now. I would say the code now looks much better in terms of its style.

**Pull request merged**